### ISSUE TITLE
LLVM>=3.5 build failure

### ISSUE THREAD

**Issue by ViralBShah** [JuliaLang/julia#13662]
I am getting a build failure on arm on both llvm 3.6.1 and llvm 3.7.0.

````
    CC src/runtime_ccall.o
/home/viral/julia/src/runtime_ccall.cpp: In function ‘jl_value_t* jl_get_cpu_name()’:
/home/viral/julia/src/runtime_ccall.cpp:138:5: error: ‘StringRef’ was not declared in this scope
     StringRef HostCPUName = llvm::sys::getHostCPUName();
     ^
/home/viral/julia/src/runtime_ccall.cpp:138:5: note: suggested alternative:
In file included from /home/viral/julia/usr/include/llvm/ADT/StringMap.h:17:0,
                 from /home/viral/julia/usr/include/llvm/Support/Host.h:17,
                 from /home/viral/julia/src/runtime_ccall.cpp:131:
/home/viral/julia/usr/include/llvm/ADT/StringRef.h:40:9: note:   ‘llvm::StringRef’
   class StringRef {
         ^
/home/viral/julia/src/runtime_ccall.cpp:138:15: error: expected ‘;’ before ‘HostCPUName’
     StringRef HostCPUName = llvm::sys::getHostCPUName();
               ^
/home/viral/julia/src/runtime_ccall.cpp:140:31: error: ‘HostCPUName’ was not declared in this scope
     return jl_pchar_to_string(HostCPUName.data(), HostCPUName.size());
                               ^
/home/viral/julia/src/runtime_ccall.cpp:141:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
make[1]: *** [runtime_ccall.o] Error 1
make: *** [julia-src-release] Error 2
````

**Comment by yuyichao** [comment]
This is not ARM specific and should be fixed by the second commit I pushed to #13655 .

**Comment by ViralBShah** [comment]
Thanks.
